### PR TITLE
10 use lib sockerio for websocket

### PR DESCRIPTION
**Pull request by totorobin**

### CODE REVIEW

File: server.mjs [hunk 1]
```diff
@@ -1,142 +1,127 @@
...
 import ViteExpress  from "vite-express";
 import http from 'http'
-import { v4 as uuidv4 } from 'uuid';
+//import { v4 as uuidv4 } from 'uuid';
+import { Server } from 'socket.io'
 
-// Our port
-let port = 8080;
 
+let port = 8080;
```
Comment (totorobin) [inline server.mjs:8]:
```suggestion
let port = process.env.PORT || 8080;
```
allow user to set port via env variable

File: server.mjs [hunk 2]
```diff
@@ -1,142 +1,127 @@
...
-            if(!Object.values(rooms[msg.roomId].users).find(u => !u.card)) {
-                notif = { text : 'Everyone has voted' }
+        }
+        socket.join(roomId)
+        io.to(roomId).emit('roomState', rooms[roomId])  // emit new state
+        socket.to(roomId).emit('joined', {name: socket.data.userName})  // inform action
+    })
+    socket.on('vote', ({value}) => {
+        socket.rooms.forEach((roomId) => {
+            if(rooms[roomId] != undefined) {
```
Comment (NatNgs) [inline server.mjs:67]:
```suggestion
            if(roomId in rooms) {
```

File: server.mjs [hunk 3]
```diff
@@ -1,142 +1,127 @@
...
-            })
-            break;
-        case 'leave' :
-            if(msg.roomId && rooms[msg.roomId]) {
-                delete rooms[msg.roomId].users[ws.id]
+        })
+    })
+    socket.on('cardVisible', (value) => {
+        socket.rooms.forEach((roomId) => {
+            if(rooms[roomId] != undefined) {
```
Comment (NatNgs) [inline server.mjs:76]:
```suggestion
            if(roomId in rooms) {
```

File: server.mjs [hunk 4]
```diff
@@ -1,142 +1,127 @@
...
+  });
+
+  io.of("/").adapter.on("delete-room", (roomId) => {
+    console.log(`room ${roomId} was deleted`);
+    delete rooms[roomId]
+  });
+  
+  io.of("/").adapter.on("leave-room", (roomId, userId) => {
+    console.log(`socket ${userId} has left room ${roomId}`);
+    if(rooms[roomId] != undefined) {
```
Comment (NatNgs) [inline server.mjs:117]:
```suggestion
    if(roomId in rooms) {
```

**Comment by totorobin** [comment]
this merge request fix more than one issue :

- [ ] #12 
- [ ] #9 
- [ ] #10 
- [ ] #1 
- [ ] #4

**Comment by totorobin** [comment]
a few more enhancement : 

- [ ] generate random room name (not just numbers)
- [ ] generate random username (star wars name)

**Pull request merged**